### ISSUE TITLE
zfs remove fails to handle removal of multiple devices

### ISSUE THREAD

**Issue by justinpryzby** [openzfs/zfs#14268]
### System information
Type | Version/Name
 --- | ---
Distribution Name	| centos
Distribution Version	| 7
Kernel Version	| 5.4.225-1.el7.elrepo
Architecture	| x86_64
OpenZFS Version	| zfs-2.1.6-1

### Describe the problem you're observing

I'm migrating data from one storage device to another.  LVM is on top of both (I know this configuration is discouraged for ZFS).
Both the old and the new devices will have 6, 128GB LVs as vdevs.  After adding the 6 new LVs to the zpool, I tried to remove five LVs from the pool to migrate off the old storage: zpool remove zfs /dev/data/zfs[45678]

But it fails like:
cannot remove /dev/data/zfs5: Pool busy; removal may already be in progress
cannot remove /dev/data/zfs6: Pool busy; removal may already be in progress
cannot remove /dev/data/zfs7: Pool busy; removal may already be in progress
cannot remove /dev/data/zfs8: Pool busy; removal may already be in progress

### Describe how to reproduce the problem

Remove multiple devices at once.

**Comment by justinpryzby** [comment]
It's still a problem in zfs-2.3.1

**Comment by amotin** [comment]
I wonder if it is a bug or a lack of feature.  Was it ever promised to work?  It would be nice if earlier vdev removal would not put data on a vdev that is going to be removed next, but honestly, I would not use device removal on such a routine scale.

**Comment by justinpryzby** [comment]
The manpage says: zpool remove [-npw] pool device…

And the --help says: remove [-npsw] <pool> <device> ...

I take that as an indication that it was supposed to work.  I'd never removed multiple devices before (and I've since realized that the it leaves behind the <indirect> device, which is ugly, so I wouldn't follow this process again).

**Comment by amotin** [comment]
Yes, I saw that in man page, but it could be written before the regular device removal was implemented.  Originally it was only for L2ARC, SLOG, etc.
